### PR TITLE
Accumulated ads and earnings disappearing after upgrade

### PR DESCRIPTION
**Pull request by tmancey**
fixes https://github.com/brave/brave-browser/issues/3716

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:

Confirm after upgrading from 0.63.4 to latest 0.63.x ad views are not lost.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/unblinded_tokens.cc [hunk 1]
```diff
@@ -64,20 +64,23 @@ bool UnblindedTokens::SetTokensFromList(const base::Value& list) {
     } else {
       base::DictionaryValue* dictionary;
       if (!value.GetAsDictionary(&dictionary)) {
-        return false;
+        DCHECK(false) << "Unblinde token should be a dictionary";
```
Comment (emerick) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/unblinded_tokens.cc]: Nit: typo

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc [hunk 2]
```diff
@@ -302,43 +311,83 @@ bool ConfirmationsImpl::GetTransactionHistoryFromDictionary(
   for (auto& transaction_value : transactions_list_value) {
     base::DictionaryValue* transaction_dictionary;
     if (!transaction_value.GetAsDictionary(&transaction_dictionary)) {
-      return false;
+      DCHECK(false) << "Transaction should be a dictionary";
+      continue;
```
Comment (emerick) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc]: It's OK to continue here when the data isn't as expected?

Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc]: yes, if transaction gets into a bad state, we should still keep good transactions

Comment (emerick) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc]: OK, got it.

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/unblinded_tokens.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/unblinded_tokens.cc]: Good spot 👍 fixed

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc]: Also this way bad transactions are still persisted in the state to help diagnose issues

**Pull request merged**